ggshield/cmd/secret/scan/changes.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+from typing import Any
+
+import click
+
+from ggshield.cmd.secret.scan.secret_scan_common_options import (
+    add_secret_scan_common_options,
+    create_output_handler,
+)
+from ggshield.cmd.utils.common_decorators import exception_wrapper
+from ggshield.cmd.utils.context_obj import ContextObj
+from ggshield.core.scan import ScanContext, ScanMode
+from ggshield.core.text_utils import display_info, pluralize
+from ggshield.utils.git_shell import (
+    check_git_dir,
+    get_default_branch,
+    get_list_commit_SHA,
+)
+from ggshield.verticals.secret.repo import scan_commit_range
+
+
+@click.command()
+@add_secret_scan_common_options()
+@click.pass_context
+@exception_wrapper
+def changes_cmd(ctx: click.Context, **kwargs: Any) -> int:
+    """
+    Scan the set of changes between the default branch and the current HEAD, including staged changes.
+    """
+    ctx_obj = ContextObj.get(ctx)
+    config = ctx_obj.config
+    check_git_dir()
+
+    default_branch = get_default_branch()
+    commit_list = get_list_commit_SHA(f"{default_branch}...HEAD")
+
+    if config.user_config.verbose:
+        display_info(
+            f"Scan staged changes and {len(commit_list)} new {pluralize('commit', len(commit_list))}"
```

Comment:
I forgot to pass `verbose` to `scan_commit_range`. Now the sha are listed, I've also updated `scan_commit_range` to print `Scanned staged changes` rather than `Scanned None` after scanning the staged changes.